code/modules/clothing/clothing.dm

```diff
@@ -66,42 +64,28 @@
 		if(M.putItemFromInventoryInHandIfPossible(src, H.held_index))
 			add_fingerprint(usr)
 
-//This code is cursed, moths are cursed, and someday I will destroy it. but today is not that day.
-/obj/item/food/clothing
-	name = "temporary moth clothing snack item"
-	desc = "If you're reading this it means I messed up. This is related to moths eating clothes and I didn't know a better way to do it than making a new food object. <--- stinky idiot wrote this"
-	bite_consumption = 1
-	// sigh, ok, so it's not ACTUALLY infinite nutrition. this is so you can eat clothes more than...once.
-	// bite_consumption limits how much you actually get, and the take_damage in after eat makes sure you can't abuse this.
-	// ...maybe this was a mistake after all.
-	food_reagents = list(/datum/reagent/consumable/nutriment/cloth_fibers = INFINITY)
-	tastes = list("dust" = 1, "lint" = 1)
-	foodtypes = CLOTH
-
-	/// A weak reference to the clothing that created us
-	var/datum/weakref/clothing
-
-/obj/item/food/clothing/make_edible()
-	. = ..()
-	AddComponent(/datum/component/edible, after_eat = CALLBACK(src, PROC_REF(after_eat)))
+//If clothing is eligible for moth consumption give it the edible component
+/obj/item/clothing/proc/make_edible()
+	if(clothing_flags & (INEDIBLE_CLOTHING | INDESTRUCTIBLE))
+		return
+	if(ispickedupmob(src))
+		return
+	var/bites = max_integrity / MOTH_EATING_CLOTHING_DAMAGE
+	AddComponent(/datum/component/edible,\
```

Comment:
it's not like this intentionally. for basically everything that checks edible, we don't want it to pass for clothes simply because moths *can* eat it.

plus, this has a non-negligible impact on init times and memory

Reply:
i should probably document this

Reply:
I'll think of another solution to this before I re-open this PR. I only went in and tried to change this because I wanted fried clothes to count as refined for moths and the current system was presenting really weird problems getting that to work because the edible component added from frying was separate from the one on the fake moth snack item.

Reply:
Easiest solution I guess is to compound on the gross current code by hardcoding in a separate check for frying when the item is clothing. blegh!

Reply:
Check the comment here: https://github.com/tgstation/tgstation/pull/73382

Reply:
Ideally "fryable" is a separate element/component or whatever. I don't think anything should be checking for just edible

Reply:
Refer to https://github.com/tgstation/tgstation/pull/73382/files#diff-b5cca44952e3b79fe2aea3038d1d705aa72e93f2b3eda7eea8f59609091ed2a8R43-R51

Reply:
Basically every item should be fryable (not just edible), and from there the moth clothes can react however they wish. Something being fryable should be completely separate from whether or not it is edible. It already should be

Reply:
I think we're talking about different things - I'm not trying to make anything fryable or not fryable, just make it so that when moth clothing is fried there's not two separate edible components now in play (the clothing and its `moth_snack`)

Reply:
I just saw that there's a signal for when something gets fried though so I'm probably just gonna use that